lib/mps_simulator.h

```diff
@@ -130,39 +136,114 @@ class MPSSimulator final {
   }
 
   void Apply1LeftOrInterior(const std::vector<unsigned>& qs,
-                            const fp_type* matrix, MPS& state) const {
+                            const fp_type* matrix, State& state) const {
     fp_type* raw_state = state.get();
     const auto bd = state.bond_dim();
     const auto l_offset = MPSStateSpace_::GetBlockOffset(state, qs[0]);
     const auto r_offset = MPSStateSpace_::GetBlockOffset(state, qs[0] + 1);
     const auto end = MPSStateSpace_::Size(state);
-    ConstOneQBMap B = ConstOneQBMap((Complex*)matrix);
-    MatrixMap C = MatrixMap((Complex*)(raw_state + end), 2, bd);
+    ConstOneQBMap B((Complex*)matrix);
+    MatrixMap C((Complex*)(raw_state + end), 2, bd);
 
     for (unsigned block_sep = l_offset; block_sep < r_offset;
          block_sep += 4 * bd) {
       fp_type* cur_block = raw_state + block_sep;
-      ConstMatrixMap A =
-          ConstMatrixMap((Complex*)(cur_block), 2, bd);
+      ConstMatrixMap A((Complex*)(cur_block), 2, bd);
       C.noalias() = B * A;
       memcpy(cur_block, raw_state + end, sizeof(fp_type) * bd * 4);
     }
   }
 
   void Apply1Right(const std::vector<unsigned>& qs, const fp_type* matrix,
-                   MPS& state) const {
+                   State& state) const {
     fp_type* raw_state = state.get();
     const auto bd = state.bond_dim();
     const auto offset = MPSStateSpace_::GetBlockOffset(state, qs[0]);
     const auto end = MPSStateSpace_::Size(state);
-    ConstOneQBMap B = ConstOneQBMap((Complex*)matrix);
-    ConstMatrixMap A =
-        ConstMatrixMap((Complex*)(raw_state + offset), bd, 2);
-    MatrixMap C = MatrixMap((Complex*)(raw_state + end), bd, 2);
+    ConstOneQBMap B((Complex*)matrix);
+    ConstMatrixMap A((Complex*)(raw_state + offset), bd, 2);
+    MatrixMap C((Complex*)(raw_state + end), bd, 2);
     C.noalias() = A * B.transpose();
     memcpy(raw_state + offset, raw_state + end, sizeof(fp_type) * bd * 4);
   }
 
+  void ApplyGate2(const std::vector<unsigned>& qs, const fp_type* matrix,
+                  State& state) const {
+    // TODO: micro-benchmark this function and improve performance.
```

Comment:
Is this TODO resolved? If not, please open an issue for it and reference that issue here.

Reply:
It is not resolved. Opened issue here: https://github.com/quantumlib/qsim/issues/373